python/tests/integration/arcticdb/test_read_batch_more.py

```diff
@@ -0,0 +1,364 @@
+    df2_0_allfilters = dataframe_filter_with_datetime_index(tmp, start1, end1)
+
+    symbol3 = "sym3" # non-existing
+
+    lib.write(symbol1, df1_0)
+    lib.update(symbol1, df1_1)
+    lib.append(symbol1, df1_2)
+    lib.update(symbol1, df1_3)
+
+    lib.write(symbol2, df2_0)
+    lib.append(symbol2, df2_1)
+    lib.append(symbol2, df2_2)
+
+    # Check Pandas update logic (simulating arctic append/update operations)
+    assert_frame_equal(df1_all, df1_3)
+    
+    # Assure last version is exactly what we expect
+    symbol1_data_sorted = lib.read(symbol1).data
+    assert_frame_equal(df1_all, symbol1_data_sorted)
+    assert_frame_equal(df1_3, symbol1_data_sorted)
+
+    # Assure previous version is what we expect
+    symbol1_data_sorted_ver_minus_one = lib.read(symbol1, as_of=1).data
+    assert_frame_equal(df1_till2, symbol1_data_sorted_ver_minus_one)
+    
+    batch = lib.read_batch(symbols=[symbol3,
+                                    symbol1, 
+                                    ReadRequest(symbol1, as_of=2), 
+                                    ReadRequest(symbol1, as_of=0),
+                                    # daterange that should produce empty DF
+                                    ReadRequest(symbol2, date_range=(dt.datetime(1990,1,1,0),dt.datetime(1999,1,1,0))), 
+                                    ReadRequest(symbol2, columns=columns_to_select),
+                                    ReadRequest(symbol2, date_range=date_range),
+                                    ReadRequest(symbol2, date_range=date_range1, columns=columns_to_select1, as_of=0)
+                                    ])
+    
+    
+    assert [vi.symbol for vi in batch] == [symbol3, symbol1, symbol1, symbol1, symbol2, symbol2, symbol2, symbol2]
+    assert isinstance(batch[0], DataError)
+    assert batch[0].symbol == symbol3
+    assert_frame_equal(df1_all, batch[1].data)
+    assert_frame_equal(df1_till3, batch[2].data)
+    assert_frame_equal(df1_0, batch[3].data)
+    assert batch[4].data.empty
+    # Only 3 colums in the DF, no date filtering
+    assert_frame_equal(df2_all_col_filtered, batch[5].data)
+    # Check if datetime filtering first several and last several is ok
+    assert_frame_equal(df2_all_without_first_and_last, batch[6].data)
+    # Column filters + datetime filters applied on the result
+    assert_frame_equal(df2_0_allfilters, batch[7].data)
+
+@pytest.mark.xfail(reason = "ArcticDB#1970")
+def test_read_batch_query_with_and(arctic_library):
+    """
+        A very small test to isolate the problem with usage of "and" 
+        in arctic queries. It produces wrong result, and should have 
+        raised an error
+    """
+
+    lib = arctic_library
+
+    symbol = "_s_"
+    df = get_sample_dataframe(100)
+
+    dfq = "bool == True and int8 > 5"
+    q = QueryBuilder()
+    q = q[q["bool"] and (q["int8"] > 5)]
+
+    lib.write(symbol, df)
+
+    batch = lib.read_batch(symbols=[ReadRequest(symbol, query_builder=q)])
```

Comment:
Is the bug reproducible only with `batch_read`? If not I think it would be better to test read or at least make parametrized test testing both read and batch_read